### PR TITLE
Update xrootd versions

### PR DESCRIPTION
**Pull request by JavierCVilla**

### CODE REVIEW

File: var/spack/repos/builtin/packages/xrootd/package.py [hunk 1]
```diff
@@ -50,3 +55,6 @@ def install(self, spec, prefix):
             cmake(source_directory, *options)
             make()
             make("install")
+
+    def url_for_version(self, version):
```
Comment (davydden) [inline var/spack/repos/builtin/packages/xrootd/package.py]: do you really need that? I think it should pick up urls just fine

Comment (JavierCVilla) [inline var/spack/repos/builtin/packages/xrootd/package.py]: You're completely right, I thought the `v ` in `v4.3.0` could cause problems during the fetch stage.
I'm deleting it.

File: var/spack/repos/builtin/packages/xrootd/package.py [hunk 2]
```diff
@@ -30,8 +30,13 @@ class Xrootd(Package):
     """The XROOTD project aims at giving high performance, scalable fault
        tolerant access to data repositories of many kinds."""
     homepage = "http://xrootd.org"
-    url      = "http://xrootd.org/download/v4.3.0/xrootd-4.3.0.tar.gz"
+    url      = "http://xrootd.org/download/v4.6.0/xrootd-4.6.0.tar.gz"
+    list_url = "http://xrootd.org/download"
```
Comment (davydden) [inline var/spack/repos/builtin/packages/xrootd/package.py]: you should not need this one as well.

Comment (JavierCVilla) [inline var/spack/repos/builtin/packages/xrootd/package.py]: Yep, removing `url_for_version` this doesn't make sense anymore

**Pull request merged**